addons/sale_management/views/sale_portal_templates.xml

```diff
@@ -14,15 +14,14 @@
         </xpath>
 
         <xpath expr="//section[@id='details']//table[@id='sales_order_table']/thead/tr" position="inside">
+            <!-- add blank Tr in thead for layout conciseness -->
             <th t-if="display_remove">
-                <!-- TODO: remove in master -->
-                <t t-set="colspan" t-value="colspan+1"/>
             </th>
         </xpath>
 
         <xpath expr="//section[@id='details']//t[@t-if='not line.display_type']" position="inside">
             <td class="text-center" t-if="display_remove">
-                <a t-att-href="sale_order.get_portal_url(suffix='/update_line/%s' % line.id, query_string='&amp;unlink=True')" class="mb8 js_update_line_json d-print-none" t-if="line.sale_order_option_ids" aria-label="Remove" title="Remove">
+                <a t-att-href="sale_order.get_portal_url(suffix='/update_line/%s' % line.id, query_string='&amp;unlink=True')" class="mb8 js_update_line_json d-print-none" t-if="sale_order and sale_order.state in ('draft', 'sent') and line.sale_order_option_ids" aria-label="Remove" title="Remove">
```

Comment:
unnecessary condition: when would `sale_order` be falsy?

Reply:
and the `display_remove` variable in the qweb template at line 13 should use the same condition (it should be false if the SO is not in state `draft` or `sent`)

Reply:
@dbo-odoo  Done